srml/support/src/storage/hashed/generator.rs

```diff
@@ -203,15 +208,45 @@ pub trait StorageValue<T: codec::Codec> {
 	///
 	/// `T` is required to implement `codec::EncodeAppend`.
 	fn append<S: HashedStorage<Twox128>, I: codec::Encode>(
-		items: &[I], storage: &mut S
-	) -> Result<(), &'static str> where T: codec::EncodeAppend<Item=I> {
+		items: &[I],
+		storage: &mut S,
+	) -> Result<(), &'static str>
+		where T: codec::EncodeAppend<Item=I>
+	{
 		let new_val = <T as codec::EncodeAppend>::append(
 			storage.get_raw(Self::key()).unwrap_or_default(),
```

Comment:
maybe the trait default should be used now here instead of type default.

let's say you have `MyStorage: Vec<u32> = vec![3]` and you do append on it, it will append from an empty vec.

Reply:
That a great point, thanks! 

Done and added a bunch of tests for it.